homeassistant/components/home_connect/sensor.py

```diff
@@ -1,29 +1,113 @@
 """Provides a sensor for Home Connect."""
 
+from dataclasses import dataclass, field
 from datetime import datetime, timedelta
 import logging
 from typing import cast
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ENTITIES
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 import homeassistant.util.dt as dt_util
 
+from .api import ConfigEntryAuth, HomeConnectDevice
 from .const import (
+    ATTR_DEVICE,
     ATTR_VALUE,
+    BSH_EVENT_PRESENT_STATE_OFF,
     BSH_OPERATION_STATE,
     BSH_OPERATION_STATE_FINISHED,
     BSH_OPERATION_STATE_PAUSE,
     BSH_OPERATION_STATE_RUN,
+    COFFEE_EVENT_BEAN_CONTAINER_EMPTY,
+    COFFEE_EVENT_DRIP_TRAY_FULL,
+    COFFEE_EVENT_WATER_TANK_EMPTY,
     DOMAIN,
+    REFRIGERATION_EVENT_DOOR_ALARM_FREEZER,
+    REFRIGERATION_EVENT_DOOR_ALARM_REFRIGERATOR,
+    REFRIGERATION_EVENT_TEMP_ALARM_FREEZER,
 )
 from .entity import HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class HomeConnectSensorEntityDescription(SensorEntityDescription):
+    """Entity Description class for binary sensors."""
+
+    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
+    options: list[str] | None = field(
+        default_factory=lambda: ["confirmed", "off", "present"]
+    )
+    state_key: str
+    appliance_types: tuple[str, ...]
+
+
+SENSORS: tuple[HomeConnectSensorEntityDescription, ...] = (
+    HomeConnectSensorEntityDescription(
+        key="Door Alarm Freezer",
+        translation_key="alarm_sensor_freezer",
+        translation_placeholders={
+            "name": "Door Alarm Freezer",
+        },
+        state_key=REFRIGERATION_EVENT_DOOR_ALARM_FREEZER,
+        appliance_types=("FridgeFreezer", "Freezer"),
+    ),
+    HomeConnectSensorEntityDescription(
+        key="Door Alarm Refrigerator",
+        translation_key="alarm_sensor_fridge",
+        translation_placeholders={
+            "name": "Door Alarm Refrigerator",
+        },
+        state_key=REFRIGERATION_EVENT_DOOR_ALARM_REFRIGERATOR,
+        appliance_types=("FridgeFreezer", "Refrigerator"),
+    ),
+    HomeConnectSensorEntityDescription(
+        key="Temperature Alarm Freezer",
+        translation_key="alarm_sensor_temp",
+        translation_placeholders={
+            "name": "Temperature Alarm Freezer",
+        },
+        state_key=REFRIGERATION_EVENT_TEMP_ALARM_FREEZER,
+        appliance_types=("FridgeFreezer", "Freezer"),
+    ),
```

Comment:
So according to icons.json, these have 2 states. Let's make this an ENUM device class sensor and add the possible states in snake_case in a list in `options`. This allows HA to know the possible values and help out with automating the state

Reply:
[3 states](https://github.com/home-assistant/core/blob/9bd5fb12b55a7029d6310a30c4b475f17db5c585/homeassistant/components/home_connect/sensor.py#L47) actually, the default state is `off`. Also, `SensorDeviceClass.ENUM` is the [default](https://github.com/home-assistant/core/blob/9bd5fb12b55a7029d6310a30c4b475f17db5c585/homeassistant/components/home_connect/sensor.py#L45) device class for this child class.

Reply:
What's confirmed?

Reply:
From the [API](https://api-docs.home-connect.com/events/#event-present-state-enumeration):
> Description: The event has been confirmed by the user.

Basically an acknowledgement of the alert by the user. It can be done at the appliance or in the Home Connect app. I plan on making a button entity for thin in HA.

Reply:
For documentation purposes -

Alarm off (Default):
<img width="323" alt="Screenshot 2024-09-11 at 9 12 57 AM" src="https://github.com/user-attachments/assets/691daa85-b096-4ae0-ad66-3f807f5b1a09">


Alarm State:
<img width="322" alt="Screenshot 2024-09-11 at 9 22 45 AM" src="https://github.com/user-attachments/assets/37eecba2-2559-4a76-9599-d3ea50a49522">


Alarm Acknowledged:
<img width="319" alt="Screenshot 2024-09-11 at 9 23 03 AM" src="https://github.com/user-attachments/assets/8695a46a-1f31-4987-9715-d30c8ea889b0">

Reply:
We should add translations for the states

Reply:
> We should add translations for the states

I think I understand now. So missing translations is why they `present` and `confirmed` state are lower case? I'll add the string translations, thanks!

Reply:
Yes, and we should move the placeholders to be translations themselves

Reply:
@joostlek, I'm seeing the translations for both the icons and states in the UI, though I'm wondering if this was not according to your recommendation. I can refactor it if you see an improvement that can be made. Thank You.